File: 0299-bulls-and-cows/0299-bulls-and-cows.java
class Solution {
    public String getHint(String secret, String guess) {
        int bulls=0,cows=0;
            int []arr=new int[10];
            for (int i = 0; i < secret.length (); i++) {
                int s=Character.getNumericValue(secret.charAt(i));
                int g=Character.getNumericValue(guess.charAt(i));
                if(s==g)bulls++;
                else{
                    if(arr[s]<0)cows++;
                    if(arr[g]>0)cows++;
                    arr[s]++;
                    arr[g]--;
                }
            }
            return bulls+"A"+cows+"B";
    }
}

File: 0299-bulls-and-cows/0299-bulls-and-cows.cpp
class Solution {
public:
    string getHint(string secret, string guess) {
         int n= secret.size();
    int m= guess.size();
    int bulls=0,cows=0;
    vector<int>g(10),s(10);
    for(int i=0;i<n;i++){
        if(secret[i]==guess[i])bulls++;
        else{
            g[guess[i]-'0']++;
            s[secret[i]-'0']++;
        }
    }
    for(int i=0;i<10;i++){
        cows +=min(g[i],s[i]);
    }
    return to_string(bulls)+"A"+to_string(cows)+"B";
    }
};